pkg/network/http/shared_libraries.go

```diff
@@ -0,0 +1,59 @@
+package http
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/network/so"
+)
+
+const (
+	libssl    = "libssl.so"
+	libcrypto = "libcrypto.so"
+)
+
+var openSSLLibs = regexp.MustCompile(
+	fmt.Sprintf("(%s|%s)", regexp.QuoteMeta(libssl), regexp.QuoteMeta(libcrypto)),
+)
+
+func findOpenSSLLibraries(procRoot string) []so.Library {
+	// all will include all host-resolved openSSL paths that alredy mapped into memory
+	libraries := so.Find(procRoot, openSSLLibs)
+
+	// we merge with the configuration set through the env vars
+	if libsFromEnv := fromEnv(); len(libsFromEnv) > 0 {
+		libraries = append(libraries, libsFromEnv...)
+	}
+
+	// prepend everything with the HOST_FS
+	// TODO: Expose this as a configuration param
+	if hostFS := os.Getenv("HOST_FS"); hostFS != "" {
```

Comment:
What is the use case for the `HOST_FS` env var?

Reply:
The `HOST_FS` represents the location where the underlying host file system is mounted. It is intended for internal use while we investigate the feasibility of the in-memory ELF parsing

Reply:
(added a comment to clarify that)